[inductor] Fix a perf regression caused by https://github.com/pytorch/pytorch/pull/98214

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98343

Summary: See the description in https://github.com/pytorch/pytorch/issues/98342

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

desertfire committed:
[inductor] Fix a perf regression caused by https://github.com/pytorch/pytorch/pull/98214

Summary: See the description in https://github.com/pytorch/pytorch/issues/98342

[ghstack-poisoned]

eellison commented on torch/_inductor/triton_heuristics.py:
```diff
@@ -138,7 +138,9 @@ def launcher({', '.join(def_args)}, grid, stream):
         launcher.n_spills = getattr(binary, "n_spills", None)
         launcher.shared = getattr(binary, "shared", None)
-        if config.triton.store_cubin:
+        launcher.store_cubin = config.triton.store_cubin
```
Why did this cause a regression if `config.triton.store_cubin` is false by default ?

desertfire replied:
Here it's not causing a problem. It is at Line 236. My theory is accessing the global `config.triton.store_cubin` is expensive when we are running multi-threading (invoking kernels through `.run`).

ngimel replied:
That's very weird, it's happening during compilation only, and shouldn't affect runtime.

desertfire commented:
@pytorchbot merge

desertfire committed:
Update on "[inductor] Fix a perf regression caused by https://github.com/pytorch/pytorch/pull/98214"


Summary: See the description in https://github.com/pytorch/pytorch/issues/98342

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "forward fix; Previous version has passed all inductor tests "
